enterprise/backend/src/metabase_enterprise/advanced_permissions/common.clj

```diff
@@ -32,12 +32,17 @@
                        :user_id api/*current-user-id* :group_id (u/the-id group-or-id)))
 
 (defn filter-tables-by-data-model-perms
-  "Given a list of tables, removes the ones for which `*current-user*` does not have data model editing permissions.
-  Returns the list unmodified if the :advanced-permissions feature flag is not enabled."
+  "Given a list of tables, removes the ones for which `*current-user*` does not have data model editing permissions."
   [tables]
-  (if (or api/*is-superuser?*
-          (not (premium-features/enable-advanced-permissions?)))
+  (cond
+    api/*is-superuser?*
     tables
+
+    ;; If advanced-permissions is not enabled, no non-admins have any data-model editing perms, so return an empty list
+    (not (premium-features/enable-advanced-permissions?))
+    (empty tables)
```

Comment:
I've basically just changed this function (and the one below) to strictly behave as its name implies. If you're an admin you have full access to tables. If you're not an admin and `advanced-permissions` isn't enabled, we don't need to do any perm checks; just return an empty list.